index.js

```diff
@@ -1,81 +1,120 @@
 // ***** Question 1 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
-
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+ console.log("%cQuestion 1", "color: red") 
+const counter = 1
+console.log(counter) 
+// => 2
+ console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+ console.log("%cQuestion 2", "color: red")
 
-// console.log(name)
-// => "Raffy"
-// name = "Not Raffy" 
-// => TypeError
-// console.log("%c----------", "color: red") 
+
+//=> "Raffy"
+const name = "menachem" 
+console.log(name)
+//=> TypeError
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+ console.log("%cQuestion 3", "color: red") 
+
+ function drinkWater(current_thirst_level){
+  console.log ("Man I sure am thirsty")
+  current_thirst_level -= 1
+  console.log ("Ahh that hits the spot")
+  return current_thirst_level
+ }
 
-// console.log(drinkWater(12))
+ console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+ console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum){
+  if (num === maybeNum){
+   return'same same'
+  }
+ else if  (num == maybeNum) {
+  return 'same same but different'
+  }
+  else{
+    return 'not same'
+  }
+}
```

Comment:
Great use of a JS if-else statement and return statements to handle this control flow. Nice job differentiating between == vs ===.